File: programme/palindrome-number/Palindrome.java
// Write a program to check palindrome number
import java.util.*;

public class Palindrome {
    public static String checkPalindrome(int a) {
        int pal = 0, n;
        int original = a;
        while (a > 0) {
            n = a % 10;
            pal = pal * 10 + n;
            a = a / 10;
        }
        if (original == pal) {
            return "Palindrome number";
        }
        return "Not palindrome number";
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Input  : ");
        int num = sc.nextInt();
        System.out.println("Output : " + checkPalindrome(num));
        sc.close();
    }
}

File: programme/palindrome-number/palindrome.js
// Write a programmeto check palindrome number

const reverseInt = function (n) {
  let remainder,
    reverse = 0;
  while (n > 0) {
    remainder = n % 10;
    reverse = reverse * 10 + remainder;
    n = Math.floor(n / 10);
  }
  return reverse;
};

const isPalindrome = function (n) {
  if (n < 0) {
    return false;
  }
  if (n < 10) {
    return true;
  }
  return reverseInt(n) == n;
};

const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

rl.question("Input  : ", (n) => {
  if (isPalindrome(n)) {
    console.log("Output : Palindrome number");
    rl.close();
  } else {
    console.log("Output : Not palindrome number");
    rl.close();
  }
});
